### PR TITLE
Cinematic Camera with bokehShader2.

### PR DESCRIPTION
**Pull request by ghost**
Changes reflected from previous [Pull request](https://github.com/mrdoob/three.js/pull/7660) as per suggestions.

Check CinematicCamera.js [here](https://github.com/kaypiKun/three.js/blob/CinematicCamera_2/examples/js/cameras/CinematicCamera.js) and the corresponding example [here](https://github.com/kaypiKun/three.js/blob/CinematicCamera_2/examples/webgl_camera_cinematic.html).

The idea was to create a simple and re-usable camera with bokeh effect, extending from Perspective Camera and using BokehShader2.

CinematicCamera behaves as PerspectiveCamera if the CinematicCamera renderer is not called in the base renderer of any example.
[Check out line 229](https://github.com/kaypiKun/three.js/blob/CinematicCamera_2/examples/webgl_camera_cinematic.html)

I hope I am not deviating from the project too much.

### CODE REVIEW

File: examples/js/cameras/CinematicCamera.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+THREE.CinematicCamera.prototype.initPostProcessing = function (){
+    if(this.postprocessing.enabled)
+    {
+        this.postprocessing.scene = new THREE.Scene();
+
+        
+        this.postprocessing.camera = new THREE.OrthographicCamera( window.innerWidth / -2, window.innerWidth / 2,  window.innerHeight / 2, window.innerHeight / - 2, -10000, 10000 );
+
+        this.postprocessing.camera.position= this.position;
+				this.postprocessing.camera.rotation= this.rotation;
```
Comment (WestLangley) [inline examples/js/cameras/CinematicCamera.js]: `position` and `rotation` are immutable. These assignments are invalid. I am not sure why these two lines are needed, anyway.

File: examples/webgl_camera_cinematic.html [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+				font-family: Monospace;
+				background-color: #f0f0f0;
+				margin: 0px;
+				overflow: hidden;
+			}
+		</style>
+	</head>
+	<body>
+
+		<script src="../build/three.js"></script>
```
Comment (WestLangley) [inline examples/webgl_camera_cinematic.html]: three.min.js

Also, title can be updated.

File: examples/webgl_camera_cinematic.html [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+				} else {
+
+					if ( INTERSECTED ) INTERSECTED.material.emissive.setHex( INTERSECTED.currentHex );
+
+					INTERSECTED = null;
+
+				}
+
+				renderer.render( scene, camera );
```
Comment (WestLangley) [inline examples/webgl_camera_cinematic.html]: I do not think this call to `render()` is required.

File: examples/js/cameras/CinematicCamera.js (hunk 1 above)
Comment (kaypiKun) [inline examples/js/cameras/CinematicCamera.js]: @WestLangley yes, that was stupid. will change

File: examples/webgl_camera_cinematic.html (hunk 3 above)
Comment (kaypiKun) [inline examples/webgl_camera_cinematic.html]: @WestLangley yes, this line was there before I implemented CinematicCamera. Will change this too...

File: examples/webgl_camera_cinematic.html (hunk 2 above)
Comment (kaypiKun) [inline examples/webgl_camera_cinematic.html]: @WestLangley title as webgl - camera cinematic is cool?

File: examples/js/cameras/CinematicCamera.js (hunk 1 above)
Comment (kaypiKun) [inline examples/js/cameras/CinematicCamera.js]: I wanted to set the position just to check, but dint needed it later. will change...

**Comment by bhouston** [comment]
This is an interesting design, but is this the way we want to go?  I am concerned that this CinematicCamera.js is limited to just one posteffect, a DOF with proper COC.  I'd rather just have an post effect that takes a camera and does the DOF COC these.

In Clara.io right now I do compute depth passes and normal passes in a central place if required by any of the post effects (such as DOF, SAO, etc.)

There is a general issue with our post effect system right now in that creating post effects that require the coordination of multiple different post effects is hard.  But I think that generally there are only a few passes that are shared, primarily depth, normal, color, ambient occlusion, and beauty.

I think that we should somehow specify in a post effect which channels it requires, such as Depth or Normal, Beauty, etc. and then the renderer will produce those channels and make them available to the requesting post effect.  This way normal and depth and other channels can be efficiently computing once, and maybe even computed as part of the primary pass if WEBGL_DrawBuffers extension is available.  Thus we can focus on what we need rather than how it is computed.  This decoupled the way those passes are computed so we can make them very efficient.

Thus while I like this example, I believe it shouldn't be in core, it should be in examples, we need a more flexible and powerful system in core.

**Comment by bhouston** [comment]
BTW I wrote up my previous comment as an issue that we could address in ThreeJS: https://github.com/mrdoob/three.js/issues/7668

**Comment by mrdoob** [comment]
@bhouston Agreed. A previous version of this PR was in core, it's now in examples. This is a nice example to have as reference and use cases to consider when improving core.

**Comment by mrdoob** [comment]
Thanks!

**Comment by kaypiKun** [comment]
Sure! I'll keep in mind the above discussion and maybe come up with something for the core...
Thanks!

**Pull request merged**